server/src-lib/Hasura/Server/Init.hs

```diff
@@ -926,13 +934,23 @@ enableAllowlistEnv =
   , "Only accept allowed GraphQL queries"
   )
 
-fallbackRefetchDelayEnv :: (String, String)
-fallbackRefetchDelayEnv =
-  ( "HASURA_GRAPHQL_LIVE_QUERIES_FALLBACK_REFETCH_INTERVAL"
-  , "results will only be sent once in this interval (in milliseconds) for "
-  <> "live queries which cannot be multiplexed. Default: 1000 (1sec)"
+planCacheSizeEnv :: (String, String)
+planCacheSizeEnv =
+  ( "HASURA_GRAPHQL_QUERY_PLAN_CACHE_SIZE"
+  , "The maximum number of query plans that can be cached, allowed values: 0-65535, " <>
+    "0 disables the cache. If this value is not set, there is no limit on the number " <>
+    "of plans that are cached"
```

Comment:
Personally, I don’t think we should default to an unlimited cache, and in fact I’m not even really sure we should allow an unbounded cache at all (since, in my opinion, an unbounded cache used in this way is more or less indistinguishable from a memory leak). I know that making informed decisions about this kind of thing is hard right now given our lack of benchmarks, but if anything, I think that’s a good reason to opt for a safe default.

In your ad-hoc tests, how significant was the performance overhead of the LRU eviction? And is it a cost imposed on every insertion, or only on evictions?

Reply:
@lexi-lambda I agree. The plan is to get rid of the unbounded cache as soon as I can spend time running benchmarks to see if there is any noticeable performance difference (will get to it as soon as my work on actions wraps up). 

> In your ad-hoc tests, how significant was the performance overhead of the LRU eviction? And is it a cost imposed on every insertion, or only on evictions?

I don't remember the exact numbers to be honest. I don't think it has anything to do with the `HashPSQ` data structure, the overhead was most likely because of the contention on the ioref which holds the cache structure as every read modifies the cache structure and it translates to a write on the ioref.